### DISCUSSION TITLE
Akai S3000XL not recognising RP2040 after firmware upgrade attempt

### DISCUSSION THREAD

**Discussion by blinduncledallas** [ZuluSCSI/ZuluSCSI-firmware#93] [Q&A]
I have a Red RP2040 that I am using in an Akai S3000XL. I purchased it from Samplerzone with its partitions already set up.

It was fine when using the firmware which it came with (v1.0.9). I tried to update to the latest RP2040-1.1 firmware and it has crashed and is not working anymore. When I powered on the S3000XL with the firmware .bin on the root it gave me rapid flashing of the led. I left it for a good while, then rebooted. Now the S3000XL doesn’t recognise the drive and there is no led activity. 

I have installed dfu-util on my MacBook but I noticed on the readme that it says ‘For RP2040-based boards, the USB programming uses .uf2 format file that can be copied to the USB drive that shows up in bootloader mode.’

Does this mean when connected to my Mac by USB, the RP2040 should show up on my Mac as a drive to simply transfer the .uf2 file? If so, this is not happening. I have also tried dfu-util but that doesn’t recognise a USB drive. 

I noticed in the zuluerr file (that I have attached) that there is a ‘HardFault’ FaultType. 

The only other thing I can think of is that the SCSI fuse may have blown in the S3000XL…

Also, I am not using the floppy power connector on the board. Will I need the extra power with this sampler? I can’t see anything about it in the manual. 

Could anyone help me to try and get this up and running again? Many thanks in advance!
[zuluerr.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/9895477/zuluerr.txt)
[zululog.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/9895478/zululog.txt)

**Comment by aperezbios** [comment]
@blinduncledallas Thanks for the report. the RP2040 itself doesn't support DFU. Instead, it uses a file format called 'uf2'. This was a design decision on the part of the people who designed the microcontroller itself.

Please download https://github.com/ZuluSCSI/ZuluSCSI-firmware/releases/download/v1.1.1-verFix/ZuluSCSI_RP2040-1.1.1-a18a7b8.uf2, then enable the BOOT DIP switch, then plug the board in to any computer via USB. Once the board is plugged in via USB, turn OFF the BOOT DIP switch. The LED will do nothing while in the bootloader mode. the RP2040 will expose a virtual USB drive, which you simply copy the .uf2 file to. Once the copy completes, the board will automatically re-set, which is why you want to have the bootloader DIP switch disabled before that happens. 

Let me know if you're able to recover using this method. It should be very straightforward.

**Reply by blinduncledallas** [reply]
Thanks for your reply.

The SCSI fuse has blown as it is not getting any power from the 3000XL. I plugged the Zulu into the floppy power and managed to update the firmware. 

Just got to wait until I change the fuse in the XL to be able to try to use it properly but I know the fuse is the problem.

I'll come back here once I've changed the fuse and if I still have problems.

**Reply by aperezbios** [reply]
@blinduncledallas great, thanks for the follow-up

**Reply by blinduncledallas** [reply]
After reading a post in the faults section about the MPC60 where someone was saying that the 1.1.1 firmware also made their board non-responsive, I perservered. I did a couple of reboots and suddenly it started working again! 

So the SCSI fuse is not blown. The 1.1.1 firmware did something to the card. Updating using the bootloader and .uf2 fixes the problem. 

The only problem I have now is I'm trying to change directories using zuluscsi.ini so I can change to folders of different samples on the 3000XL when I need to. If I can solve this I will be over the moon!

I have set the right drive naming (HD00_512xxx.img etc) within the folders. For example, I have a folder named "Set_1" which I would like to boot to. I set this as #Dir = "/Set_1" in zuluscsi.ini which is in plain text but the 3000XL only recognises files on the root of the SD card no matter what I try. 

It's probably something I am doing wrong or I haven't set up the zuluscsi.ini file properly. I am learning on the fly.

My 3000XL drive ID is 6, the ZuluSCSI ID is 5. This is the setting in the 3000XL. I keep 6 clear in the naming of the drive ID's on the SD card. 

I have attached the zuluscsi.ini I am using. Could you possibly look at it and tell me where I am going wrong?
 
[zuluscsi.ini.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/9946828/zuluscsi.ini.txt)

**Reply by aperezbios** [reply]
@blinduncledallas if you haven't removed the hash mark from the beginning of the Dir line, then that would be the reason why it's not being honored. Any text behind # is ignored by the config file parser, by design. The zuluscsi.ini file you provided as an attachment still has this line commented out.
